### ISSUE TITLE
Restored geometry increases with every restart in windowed mode

### ISSUE THREAD

**Issue by JulienMaille** [antonypro/QGoodWindow#21]
I've pushed a reproducer here: https://github.com/JulienMaille/QGoodWindow
Every time the application is restarted the width is increased by 32px and the height by 6px

I discovered that you had to reimplement restoreGeometry (!) and followed the startup, what I see is the following

## restoreGeometry
When closing the app, we store following geometry in the settings
`QGoodWindow::normalGeometry returned { x = 2203, y = 103, width = 626, height = 642 }    QRect`

When resuming the app, we read from the settings
`normal_geometry { x = 2203, y = 103, width = 626, height = 642 }    QRect`

`QGoodWindow::restoreGeometry` will call `QGoodWindow::resize(626, 642)` which will increase it to **(642, 650)**  (+16, +8)

`QGoodWindow::restoreGeometry` will then call `QGoodWindow::showNormal()` then `setGeometry(normalGeometry())` which will call again `QGoodWindow::resize(642, 650)` and again increase it to **(658, 658)**  (+16, +8)

If we skip one of the 2 size increase (I tried both) we fix on side of the issue.

## saveGeometry
Here I believe we should compensate the size increase due to border like this
```c++
    QRect ng = normalGeometry();

    if( m_win_use_native_borders )
    {
        const int border_width = BORDERWIDTH;
        ng.setWidth(ng.width() - border_width * 2);
        ng.setHeight(ng.height() - border_width);
    }

    stream << ng;
    stream << bool(windowState() & Qt::WindowMinimized);
    stream << bool(windowState() & Qt::WindowMaximized);
    stream << bool(windowState() & Qt::WindowFullScreen);
```

**Comment by JulienMaille** [comment]
Removing the sizeincrease in `void QGoodWindow::resize(int width, int height)` also fixes the problem but I suppose there's a reason for that bit of code although I could not see any obvious side effects after removal

**Comment by antonypro** [comment]
@JulienMaille: This summing in necessary because on Windows 11 the left, right and bottom borders are native borders, this allow native behavior of QGoodWindow including Windows 11 rounded corners.

And a bug that i found in my code is set `frameGeometry();` to `m_rect_normal` variable. To fix this issue, is needed to change it to `geometry();` [here](https://github.com/antonypro/QGoodWindow/blob/02225cdc922787fab54e6f8ee48d24b85b67be27/QGoodWindow/QGoodWindow/src/qgoodwindow.cpp#L3002). Then the proper calculations for geometry is used and the window does not grow up each close/reopen.

Can you please confirm this? If this is the correct the solution it will be added in next version.

Thanks!

**Comment by JulienMaille** [comment]
> And a bug that i found in my code is set `frameGeometry();` to `m_rect_normal` variable. To fix this issue, is needed to change it to `geometry();` [here](https://github.com/antonypro/QGoodWindow/blob/02225cdc922787fab54e6f8ee48d24b85b67be27/QGoodWindow/QGoodWindow/src/qgoodwindow.cpp#L3002).

I confirm it works with that fix

**Comment by antonypro** [comment]
@JulienMaille: Thanks for the report, this fix will be added in next version.

**Comment by antonypro** [comment]
This bug was solved on version 2.2.
